addons/visiblefavor/visiblefavor.lua

```diff
@@ -0,0 +1,130 @@
+    [0x1c] = 0x66, -- Cait Sith
+}
+
+refresh = true
+pet_index = 0
+favor_buff = false
+
+defaults = {}
+defaults.enabled = true
+defaults.display = 'favor'
+defaults.effect = 'debuff'
+
+settings = config.load(defaults)
+
+function check_incoming_chunk(id, original, modified, injected, blocked)
+    if id == 0x037 then
+        refresh = true
+    elseif settings.enabled and id == 0x00E then
+        if refresh and settings.display ~= 'all' then
+            local player = windower.ffxi.get_player()
+            pet_index = windower.ffxi.get_mob_by_index(player.index).pet_index
+            if settings.display == 'favor' then
+                for _,buff_id in pairs(player.buffs) do
+                    if buff_id == 431 then
+                        favor_buff = true
+                        break
+                    end
+                    favor_buff = false
+                end
+            end
+        end
+        local npc_index = original:unpack('H', 9)
+        if settings.display == 'all' or settings.display == 'self' and pet_index == npc_index or settings.display == 'favor' and favor_buff and pet_index == npc_index then
+            local npc_model = original:unpack('H', 0x33)
+            if model_to_effect[npc_model] then
+                local effect = model_to_effect[npc_model]
+                if settings.effect == 'debuff' then
+                    effect = effect + 8
+                end
+                return modified:sub(1, 38) .. string.char(effect) .. modified:sub(40)
+            end
+        end
+    end
+end
+
+function visiblefavor_command(...)
+    local arg = {...}
+    for k,v in pairs(arg) do
+        arg[k] = arg[k]:lower()
+    end
+    if #arg == 1 and arg[1] == 'toggle' then
+        settings.enabled = not settings.enabled
+    elseif #arg == 1 and arg[1] == 'on' then
+        settings.enabled = true
+    elseif #arg == 1 and arg[1] == 'off' then
+        settings.enabled = false
+    elseif #arg == 2 and arg[1] == 'display' and arg[2] == 'all' then
+        settings.display = 'all'
+    elseif #arg == 2 and arg[1] == 'display' and arg[2] == 'self' then
+        settings.display = 'self'
+    elseif #arg == 2 and arg[1] == 'display' and arg[2] == 'favor' then
+        settings.display = 'favor'
+    elseif #arg == 2 and arg[1] == 'effect' and arg[2] == 'buff' then
+        settings.effect = 'buff'
+    elseif #arg == 2 and arg[1] == 'effect' and arg[2] == 'debuff' then
+        settings.effect = 'debuff'
+    else
+        windower.add_to_chat(167, 'Command usage:')
+        windower.add_to_chat(167, '    vf toggle/on/off')
+        windower.add_to_chat(167, '    vf display all/self/favor')
+        windower.add_to_chat(167, '    vf effect buff/debuff')
+        return
+    end
+    windower.add_to_chat(207, 'VisibleFavor: enable = %s, display = \031\200%s\030\001, effect = \031\200%s\030\001':format(settings.enabled and '\031\204yes\030\001' or '\031\167no\030\001', settings.display, settings.effect))
```

Comment:
I tried the hex notation and it didn't work. I looked it up and it seems like that was something added in Lua 5.2, which I assume Windower is still using 5.1? Either way, I switched to the full "\ddd" notation on them instead of the shortened form. Not ideal but better then before.